src/platformdirs/macos.py

```diff
@@ -33,7 +24,7 @@ def site_data_dir(self) -> str:
     @property
     def user_config_dir(self) -> str:
         """:return: config directory tied to the user, e.g. ``~/Library/Preferences/$appname/$version``"""
-        return self._path_with_app_name_version("~/Library/Preferences/", expand=True)
+        return self._path_with_app_name_version(os.path.expanduser("~/Library/Preferences/"))
```

Comment:
Why not move `os.path.expanduser` into `_path_with_app_name_version`?

Reply:
We don't want to do it when we don't have to do it, because it's not a cheap operation. And @RonnyPfannschmidt just argued against hiding it behind a boolean flag.

Reply:
That makes sense.

Maybe an extra method would make sense here:

```python
def _expand_and_append_app_name_and_version(self, *base: str) -> str:
    return os.path.expanduser(self._path_with_app_name_version(*base))
```

To avoid so much repetition.

Reply:
I prefer the way it's now 👍